docs/adr/0000-use-toolforge-kubernetes-for-test-system.md

```diff
@@ -0,0 +1,28 @@
+# 0) Use Toolforge kubernetes infrastructure for the test system
+
+Date: 2020-09-25
+
+## Status
+
+proposed
+
+## Context
+
+This project is new and its exact technological future is unknown, for example, we still don't know if this will have to be nodejs app in future and it will be only client-side app being built from npm modules and served as static assets using Apache, Nginx, etc.
+
+We also need a publicly accessible test setup to have its features tested by UX and PM once they land on the main branch. We also need a development environment, need a CI environment and so on. Each environment has its own unique requirements, for production we are not sure we would use pipeline as a node app or it will be part of the standard WDQS GUI assets.
+
+Toolforge seems like a good setup for small projects due to its baked-in support of serving static assets using lighttpd or serving node10 apps. It has been changing and growing more mature. One of its most recent features is support for dockerized webservices and cronjobs using kubernetes. You can read more about it in [here](https://wikitech.wikimedia.org/wiki/Help:Toolforge/Kubernetes). We can't use our built Dockerfile in Toolforge and if we want to go in that direction, we would have to have a dedicated VM just for this service in CloudVPS instead and make it automatically updated using puppet or ansible which would be too much work for too little gain.
```

Comment:
I can ask around (and I couldn't find such documentation) but I assume it's because any arbitrary Dockerfile gives a big degree of freedom which could become problematic (noisy neighbour effect).

Reply:
The answer I got in IRC:
> Amir1: you would then need to upload the resulting docker img to our toolforge registry, right? We don't support bring-your-own-container yet, even if it is based on the WMF docker registry